### DISCUSSION TITLE
Put the rain gauge into operation

### DISCUSSION THREAD

**Discussion by geraldFS** [arendst/Tasmota#11445] [Ideas]
Hi,

I'm still tinkering with my weather station, the wind direction display works thanks to your help!

Now I'm still on the rain display or a qualitative display of the amount of rain.
The mechanics are relatively simple, a seesaw in which the rainwater is collected, when the level is reached, it tilts and empties the water.
A reed contact is switched with every tilt.

I have now given a counter on the pin but it does not count 1 - 2 - 3 etc. up but 9 - 14 - 21 etc. up but no logic is recognizable to me.

Is there a possibility that one can somehow implement the following scenarios.

a) count up evenly
b) Conversion in mm / m² (each signal = 0.2794 mm / m²
c) Evaluation and display of the amount of rain for the last 1 hour / 6h / 24h / 7 days
d) Send the results from point C as an MQTT message

It would be great if there could be a solution how to implement this scenario, many are trying to solve this problem on the internet.

It is tried with scripts based on SML, but somehow everything is not so good, or I can not do it.

It would be perfect if there was a finished module ;-)
the anemometer (xsns_68_windmeter.ino) works wonderfully.

Thanks a million for your help!

Gerald

**Comment by barbudor** [comment]
> I have now given a counter on the pin but it does not count 1 - 2 - 3 etc. up but 9 - 14 - 21 etc. up but no logic is recognizable to me.

I would guess your reed switch requires debouncing. So you will have to play with [CounterDebounce](https://tasmota.github.io/docs/Commands/#counterdebounce) , CounterDebounceLow and CounterDebounceHigh.
I would expect your reed switch to make a small pulse when passing by the vertical.
1) Is that a pulse to VCC ___/\___  or a pulse to GND ----v---- ?
2) You need to find out the duration of the pulse
3) and the max frequency of the tilting to know the max duration of the resting state
Then play with the above commands to find suitable setting

you may also add a small RC filter (resistance in serie with the signal from the reed and a capacitor across the signal  after the resistance and the VCC or GND depending on which your reed is connected (VCC is reed connected to VCC, GND if reed connected to GND)

This should help the counting to be correct

b) You need to know the capacity of the buckets to make this transformation. You will need RULES + EXPRESSION to do some calculations
d) Once b) is solved, easy to use Publish

c) Display new information on the web gui requires [Scripting-Language](https://tasmota.github.io/docs/Scripting-Language)
You can make graphs  using Googlegraphs but I'm no expert on that
I would most strongly advise to stop at a+b+d and do all the rest on a PC backend
Whether it's an tool such as HA, OpenHAB, NodeRed or InfluxDB + Grafana

**Comment by geraldFS** [comment]
i connected it to D0 with a PullUP 4.7k resistor.

**Reply by barbudor** [reply]
Should I guess that the other side of the reed is connected to GND ?
When dealing with electronics, schematics are a useful tool to explain others what you've done instead of letting them guess your cabling.

However D0 = GPIO16 seems to be a bad choice for counters as counters requires interrupts and GPIO16 do not support interrupt
![image](https://user-images.githubusercontent.com/2996042/112210318-d5446600-8c1a-11eb-82ec-8d88f72acbd0.png)

**Comment by geraldFS** [comment]
Sure, sorry here is the basic circuit diagram.

I quickly installed an RC element, but now it's better but not quite good. Will increase the capacitor again tomorrow.

Gerald

![Tasmota](https://user-images.githubusercontent.com/35431341/112212550-6ae0f500-8c1d-11eb-8d56-c341e6681c08.png)
![Rain Gauge](https://user-images.githubusercontent.com/35431341/112212572-703e3f80-8c1d-11eb-87dd-693720f6f9ae.png)


Between update, I have now put hn on the D7 with a 4.7k pull-up resistor no change.

**Reply by barbudor** [reply]
I would suggest more something like this
![image](https://user-images.githubusercontent.com/2996042/112214600-ab417280-8c1f-11eb-986c-aa19fbfba41f.png)

The value of 300 ohms and 100nF are indicative. You may have to tweak .
As Tasmota will count the falling edges, you need to have a short discharging time and a slighly longer charging time.
So when the reed close (possibly multiple times when the magnet pass by), the signal will drop low quickly and remains low for the whole duration of the pulse.
A digital sampling oscilloscope would be useful to see the form of the wave.
